webextension/selector/lifecycle.js

```diff
@@ -0,0 +1,51 @@
+/* globals loadSelector, catcher */
+
+window.lifecycle = (function () {
+  let exports = {};
+  if (window.lifecycle) {
+    window.lifecycle.unload();
+  }
+  document.dispatchEvent(new Event("pageshot-unload"));
+
+  exports.unload = function () {
```

Comment:
More a nit than problem... This unload function doesn't follow the pattern of cleaning up itself, it cleans up everything, so it might be better to name it something else. Then we wouldn't have to special case it below.